keras/src/layers/preprocessing/spectrogram_test.py

```diff
@@ -0,0 +1,196 @@
+            window=window_arr,
+            nperseg=frame_length,
+            noverlap=frame_length - frame_step,
+            mode=mode,
+            scaling=scaling,
+            detrend=False,
+            nfft=fft_length,
+        )
+        Y_true = np.transpose(S, [0, 2, 1])
+        return Y_true, Y
+
+    @pytest.mark.requires_trainable_backend
+    def test_spectrogram_basics(self):
+        self.run_layer_test(
+            layers.Spectrogram,
+            init_kwargs={
+                "frame_length": 500,
+                "frame_step": 25,
+                "fft_length": 1024,
+            },
+            input_shape=(2, 16000, 1),
+            expected_output_shape=(2, 15500 // 25 + 1, 513),
+            expected_num_trainable_weights=2,
+            expected_num_non_trainable_weights=0,
+            expected_num_seed_generators=0,
+            expected_num_losses=0,
+            supports_masking=False,
+        )
+
+        self.run_layer_test(
+            layers.Spectrogram,
+            init_kwargs={
+                "frame_length": 150,
+                "frame_step": 71,
+                "fft_length": 4096,
+                "mode": "real",
+            },
+            input_shape=(2, 160000, 1),
+            expected_output_shape=(2, 159850 // 71 + 1, 2049),
+            expected_num_trainable_weights=1,
+            expected_num_non_trainable_weights=0,
+            expected_num_seed_generators=0,
+            expected_num_losses=0,
+            supports_masking=False,
+        )
+
+        self.run_layer_test(
+            layers.Spectrogram,
+            init_kwargs={
+                "frame_length": 150,
+                "frame_step": 43,
+                "fft_length": 512,
+                "mode": "imag",
+                "padding": "same",
+            },
+            input_shape=(2, 160000, 1),
+            expected_output_shape=(2, 160000 // 43 + 1, 257),
+            expected_num_trainable_weights=1,
+            expected_num_non_trainable_weights=0,
+            expected_num_seed_generators=0,
+            expected_num_losses=0,
+            supports_masking=False,
+        )
+        self.run_layer_test(
+            layers.Spectrogram,
+            init_kwargs={
+                "frame_length": 150,
+                "frame_step": 10,
+                "fft_length": 512,
+                "trainable": False,
+                "padding": "same",
+            },
+            input_shape=(2, 160000, 1),
+            expected_output_shape=(2, 160000 // 10 + 1, 257),
+            expected_num_trainable_weights=0,
+            expected_num_non_trainable_weights=2,
+            expected_num_seed_generators=0,
+            expected_num_losses=0,
+            supports_masking=False,
+        )
+
+    @pytest.mark.requires_trainable_backend
+    def test_spectrogram_error(self):
+        rnd = np.random.RandomState(41)
+        X = rnd.uniform(low=-1, high=1, size=(4, 160000, 1)).astype(np.float64)
```

Comment:
Use lowercase variable names

Reply:
Renamed all the uppercase variables in the test module to lowercase.